src/coreclr/gc/gc.cpp

```diff
@@ -18218,6 +18243,14 @@ int gc_heap::generation_to_condemn (int n_initial,
         }
     }
 
+#ifdef USE_REGIONS
+    if (!try_get_new_free_region())
+    {
+        dprintf (GTC_LOG, ("can't get an empty region -> full compacting"));
```

Comment:
In your comments for the PR, you said in this case we will do a special kind of sweep. The dprintf says we do a full compacting GC. Is there a contradiction, or what am I missing?

Reply:
it's the "and find out that we do need it," part - we try to get a new region here just to prepare for *if* we need it, as long as we don't actually need it, we will be doing a full compacting GC. but if we can't get it here and later find that we actually need it that's when we absolute run out of memory and we only do the special sweep then.

Reply:
Got it, thanks!